pkg/common/security/usergroup_ldap_resolver.go

```diff
@@ -0,0 +1,229 @@
+	files, err := os.ReadDir(secretsDir)
+	if err != nil {
+		log.Log(log.Security).Error("Failed to read secrets directory",
+			zap.Error(err))
+		return
+	}
+
+	secretCount := 0
+	// Iterate over all secret files in the secrets directory
+	for _, file := range files {
+		fileName := file.Name()
+
+		// Skip non-secret entries such as Kubernetes internal metadata (e.g., symlinks like "..data" or directories like "..timestamp")
+		if strings.HasPrefix(fileName, "..") || file.IsDir() {
+			log.Log(log.Security).Info("Ignoring non-secret entry (Kubernetes metadata entry or directory)",
+				zap.String("name", fileName))
+			continue
+		}
+
+		secretCount++
+		secretKey := fileName // use the cached fileName
+		secretValueBytes, err := os.ReadFile(filepath.Join(secretsDir, secretKey))
+		if err != nil {
+			log.Log(log.Security).Warn("Could not read secret file",
+				zap.String("file", secretKey),
+				zap.Error(err))
+			continue
+		}
+		secretValue := strings.TrimSpace(string(secretValueBytes))
+
+		// log the secret key and value
+		log.Log(log.Security).Debug("Loaded LDAP secret",
+			zap.String("key", secretKey))
+
+		switch secretKey {
+		case common.LdapHost:
+			ldapConf.Host = secretValue
+		case common.LdapPort:
+			ldapConf.Port, err = strconv.Atoi(secretValue)
+			if err != nil {
+				log.Log(log.Security).Warn("Failed to convert LDAP port to integer, using default port", zap.Error(err))
+				ldapConf.Port = common.DefaultLdapPort
+			}
+		case common.LdapBaseDN:
+			ldapConf.BaseDN = secretValue
+		case common.LdapFilter:
+			ldapConf.Filter = secretValue
+		case common.LdapGroupAttr:
+			ldapConf.GroupAttr = secretValue
+		case common.LdapReturnAttr:
+			ldapConf.ReturnAttr = strings.Split(secretValue, ",")
+		case common.LdapBindUser:
+			ldapConf.BindUser = secretValue
+		case common.LdapBindPassword:
+			ldapConf.BindPassword = secretValue
+		case common.LdapInsecure:
+			ldapConf.Insecure, err = strconv.ParseBool(secretValue)
+			if err != nil {
+				log.Log(log.Security).Warn("Failed to convert LDAP Insecure to boolean, using default value", zap.Error(err))
+				ldapConf.Insecure = common.DefaultLdapInsecure
+			}
+		case common.LdapSSL:
+			ldapConf.SSL, err = strconv.ParseBool(secretValue)
+			if err != nil {
+				log.Log(log.Security).Warn("Failed to convert LDAP SSL to boolean, using default value", zap.Error(err))
+				ldapConf.SSL = common.DefaultLdapSSL
+			}
+		default:
+			log.Log(log.Security).Warn("Encountered unrecognized LDAP secret key",
+				zap.String("key", secretKey))
+		}
+	}
+	log.Log(log.Security).Info("Finished loading LDAP secrets",
+		zap.Int("numberOfSecretsLoaded", secretCount))
+}
+
+func GetUserGroupCacheLdap() *UserGroupCache {
+	readSecrets()
+
+	return &UserGroupCache{
+		ugs:           map[string]*UserGroup{},
+		interval:      cleanerInterval * time.Second,
+		lookup:        LdapLookupUser,
+		lookupGroupID: LdapLookupGroupID,
+		groupIds:      LDAPLookupGroupIds,
```

Comment:
Do we need to initialize `stop` channel as well? We might to add test related based on cache clean up, interval set up etc for this new resolver